### PR TITLE
Fix Doc Building

### LINKED ISSUE #196: Dev Documentation not up-to-date
I noticed that our dev documentation is not up-to-date. It appears, that since the avalanche integration building no longer works. Our test failed to spot it since it was installing dependencies differently to how RTD is doing it.

I'm working on rectifying the test and fixing all issues we've added over the time as part of this PR: #195

### PR DESCRIPTION
**Pull request by wistuba**
Our doc building action installs Renate[dev] while rtd does not. Therefore, our tests are passing but the docs are not updating anymore. This PR fixes the github action to reflect what rtd is doing and a hotfix addresses the problem.

Closes #196 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by wesk** [comment]
For Syne Tune I just followed this: https://docs.readthedocs.io/en/stable/guides/git-integrations.html#provider-specific-instructions

No code-changes needed, wondering if something similar can be done for Renate

**Comment by wistuba** [comment]
This is not about updating the files on rtd but on catching errors in the doc building process. The reason rtd is not updating the files because the building is failing.

File: src/renate/updaters/avalanche/plugins.py [hunk 1]
```diff
@@ -19,11 +19,6 @@ def _make_checkpoint_file_path(self, checkpoint_name: str) -> Path:
     def checkpoint_exists(self, checkpoint_name: str) -> bool:
         return self._make_checkpoint_file_path(checkpoint_name).exists()
 
-    def load_checkpoint(self, checkpoint_name: str, checkpoint_loader) -> Any:
-        checkpoint_file = self._make_checkpoint_file_path(checkpoint_name)
-        with open(checkpoint_file, "rb") as f:
-            return checkpoint_loader(f)
-
```
Comment (610v4nn1) [inline src/renate/updaters/avalanche/plugins.py:26]: This change has nothing to do with sphinx, is it intended?

**Review by 610v4nn1 — APPROVED** [review]

File: src/renate/updaters/avalanche/plugins.py (hunk 1 above)
Comment (wistuba) [reply src/renate/updaters/avalanche/plugins.py:26]: not sure why but sphinx will complain about that code. it is also identical to the parent's code, so we can drop it: https://github.com/ContinualAI/avalanche/blob/v0.3.1/avalanche/training/plugins/checkpoint.py#L335

**Pull request merged**